dali/python/nvidia/dali/pipeline.py

```diff
@@ -920,6 +920,14 @@ def schedule_run(self):
         Should not be mixed with :meth:`run` in the same pipeline"""
         with self._check_api_type_scope(types.PipelineAPIType.SCHEDULED):
             if self._first_iter and self._exec_pipelined:
+                # For one, when prefetching, parallel external source will reuse buffers
+                # that might be still referenced by no_copy input fed to pipeline
+                if not self.empty():
+                    warnings.warn(
```

Comment:
I think it is possible, prefetch queue depth is 2, batches to consume is 1, we can still schedule one more run. The native part can overschedule - it will just wait for the empty output buffer, but the ES may fail (parallel mode with nocopy, but the regular ES does copy and have an internal queue).

Reply:
Do you mean that it should be supported? By possible you mean that using fw iterators correctly you may still end up in such situation?

Reply:
> Should we warn only if we have PES and treat it as extra limitation of schedule api that is not there otherwise.

I think so. You can still use this API without the FW iterator.